### ISSUE TITLE
FuncMap refactorization

### ISSUE THREAD

**Issue by lmoffatt** [lmoffatt/macro_dr#73]
So: here are the changes

1. Memoization, Time_it and Test_it will become single threaded. 
2. F_on_thread dissapears
3. FuncMap gets F_on_thread functionality (method f)
4. Two new methods are created fork (n)-> creates a vector of FuncMap and 
operator+= vector<FuncMap> with the corresponding operators += for each Function wrapper.

**Comment by lmoffatt** [comment]
Time_it_st is the single threaded implementation of Time_it.  commit 9bd47dd

**Comment by lmoffatt** [comment]
Now we should see how Memoization is used. 

Lets see: 
```C++
var::Thread_Memoizer(
          var::F(Calc_Qdt_step{},
                 [](auto &&...x) {
                   auto m = Macro_DMR{};
                   return m.calc_Qdt_ATP_step(std::forward<decltype(x)>(x)...);
                 }),
          var::Memoiza_all_values<Maybe_error<Qdt>, ATP_step, double>{},
          num_scouts_per_ensemble / 2),
   
```

**Comment by lmoffatt** [comment]
so, we have two options: 
Option one we remove the thread function of Thread_Memoizer. 
Option two we make the Memoiza_all_values to include the function. 

I guess the first option is better. 
So we create Single_Thread_Memoizer

**Comment by lmoffatt** [comment]
Test_it is not implemented (is right now just a copy of Time_it)

**Comment by lmoffatt** [comment]
Now I implement FuncMapSt

**Comment by lmoffatt** [comment]
Now the problem is how to integrate the new functionality into the old code. 

1. build a new get_Function_Table-->get_Function_TableSt (561e47b)
2.

**Comment by lmoffatt** [comment]
The solution is simple: 
make both interfaces or FuncMap and FuncMapSt temporarilty the same in this way we can make the FuncMapSt code to compile and then by removing the temporarily changes we remove FuncMap altogether

**Comment by lmoffatt** [comment]
The first part is to integrate FuncMapSt into existing code.

**Comment by lmoffatt** [comment]
Run it, and it gave weird results (waaay too many min violation errors). 
It turned out that the logic of Memoization clearance was flawed. 
Solved in commit 678bc74
